src/wui/interactive_base.cc

```diff
@@ -750,6 +750,15 @@ void InteractiveBase::draw_overlay(RenderTarget& dst) {
 
 	Game* game = dynamic_cast<Game*>(&egbase());
 
+	if (in_road_building_mode() && tooltip().empty()) {
+		draw_tooltip(
+		   in_road_building_mode(RoadBuildingType::kRoad) ?
+		      (boost::format(_("Road length: %u")) % get_build_road_path().get_nsteps()).str() :
+		      (boost::format(_("Waterway length: %1$u (limit: %2$u)")) %
```

Comment:
I think we can make this one a bit more concise:

```suggestion
		      (boost::format(_("Waterway length: %1$u/%2$u")) %
```

Let's keep the road length.